docker/lookaside_test.go

```diff
@@ -75,11 +91,35 @@ func TestRegistriesDirPath(t *testing.T) {
 				RootForImplicitAbsolutePaths: rootPrefix,
 				RegistriesDirPath:            nondefaultPath,
 			},
+			false,
+			nondefaultPath,
+		},
+		// User registries.d overridden
+		{&types.SystemContext{}, true, userRegistriesDirPath},
+		// Context and user User registries.d preset simultaneously
+		{&types.SystemContext{RegistriesDirPath: nondefaultPath}, true, nondefaultPath},
+		// Root and user registries.d overrides present simultaneously,
+		{
+			&types.SystemContext{
+				RootForImplicitAbsolutePaths: rootPrefix,
+				RegistriesDirPath:            nondefaultPath,
+			},
+			true,
 			nondefaultPath,
 		},
 		// No environment expansion happens in the overridden paths
-		{&types.SystemContext{RegistriesDirPath: variableReference}, variableReference},
+		{&types.SystemContext{RegistriesDirPath: variableReference}, false, variableReference},
 	} {
+		if c.userFilePresent {
+			err := os.MkdirAll(filepath.Dir(userRegistriesDirPath), os.ModePerm)
+			require.NoError(t, err)
+			f, err := os.Create(userRegistriesDirPath)
+			require.NoError(t, err)
+			f.Close()
+		} else {
+			os.Remove(userRegistriesDirPath)
+			require.NoError(t, err)
```

Comment:
`err` is not set in the `os.Remove` call above, this is checking the result of `Setenv` above or something like that.

But if two non-`userFilePresent` tests exist in a sequence, this should not fail on “does not exist” anyway. So, either use `os.RemoveAll`, or check the error here, but only if it is not “directory does not exist”.